src/bbs2gh/Handlers/MigrateRepoCommandHandler.cs

```diff
@@ -369,15 +387,17 @@ private void ValidateDownloadOptions(MigrateRepoCommandArgs args)
 
         if (shouldUseSsh && shouldUseSmb)
         {
-            throw new OctoshiftCliException("Either SSH or SMB download options can be specified.");
+            throw new OctoshiftCliException("You can't provide both SSH and SMB credentials together.");
         }
 
-        if (shouldUseSsh && sshArgs.Any(arg => arg.IsNullOrWhiteSpace()))
+        if ((args.SshUser.HasValue() && args.SshPrivateKey.IsNullOrWhiteSpace()) ||
+            (args.SshPrivateKey.HasValue() && args.SshUser.IsNullOrWhiteSpace()))
```

Comment:
We can use XOR here:
```suggestion
        if (args.SshUser.HasValue() ^ args.SshPrivateKey.HasValue())
```